src/repr/src/strconv.rs

```diff
@@ -606,6 +607,21 @@ where
     Nestable::MayNeedEscaping
 }
 
+pub fn parse_pg_legacy_name(s: &str) -> String {
```

Comment:
What do we do if we end up with collisions? Is this something we ignore for now?

Reply:
I think this responsibility should be up to the caller of this function. We don't actually use this to parse object identifiers so I think it's fine.